exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpExporterBuilder.java

```diff
@@ -125,33 +126,84 @@ public HttpExporterBuilder<T> exportAsJson() {
     return this;
   }
 
-  public HttpExporter<T> build() {
-    Map<String, String> headers = this.headers == null ? Collections.emptyMap() : this.headers;
-    Supplier<Map<String, String>> headerSupplier = () -> headers;
-
-    HttpSender httpSender = null;
-    // TODO: once we publish multiple HttpSenderProviders, log warning when multiple are found
-    for (HttpSenderProvider httpSenderProvider :
+  /**
+   * Resolve the {@link HttpSenderProvider}.
+   *
+   * <p>If no {@link HttpSenderProvider} is available, throw {@link IllegalStateException}.
+   *
+   * <p>If only one {@link HttpSenderProvider} is available, use it.
+   *
+   * <p>If multiple are available and..
+   *
+   * <ul>
+   *   <li>{@code io.opentelemetry.exporter.internal.http.HttpSenderProvider} is empty, use the
+   *       first found.
+   *   <li>{@code io.opentelemetry.exporter.internal.http.HttpSenderProvider} is set, use the
+   *       matching provider. If none match, throw {@link IllegalStateException}.
+   * </ul>
+   */
+  private static HttpSenderProvider resolveHttpSenderProvider() {
```

Comment:
tiny nit: would prefer if this method were below the calling method, to make the public `build` method more prominent in the file.